pydantic/main.py

```diff
@@ -151,7 +151,7 @@ def validate_custom_root_type(fields: Dict[str, Field]) -> None:
 
 
 class MetaModel(ABCMeta):
-    @no_type_check
+    @no_type_check  # noqa C901
```

Comment:
@samuelcolvin for the sake of diff-legibility, I didn't worry about refactoring this method to reduce complexity yet. But let me know if you have a preference over what to pull out into a separate function, or if you'd just prefer to keep the noqa C901

Reply:
I think this is fine for now. Definitely refactoring should be a different PR if required at all.